docs/make.jl

```diff
@@ -10,6 +10,43 @@ if haskey(ENV, "DOCSARGS")
     end
 end
 
+# ==============================================================================
+#  Modify the release notes
+# ==============================================================================
+
+function fix_release_line(
+    line::String,
+    url::String = "JuliaDocs/Documenter.jl",
+)
+    # (abc#XXXX) -> ([abc#XXXX](abc/issue/XXXX))
+    while (m = match(r"\(([a-zA-Z0-9/]+?)\#([0-9]+)\)", line)) !== nothing
+        new_url, id = m.captures[1], m.captures[2]
+        line = replace(line, m.match => "([$new_url#$id](https://github.com/$new_url/issues/$id))")
+    end
+    # (#XXXX) -> ([#XXXX](url/issue/XXXX))
+    while (m = match(r"\(\#([0-9]+)\)", line)) !== nothing
+        id = m.captures[1]
+        line = replace(line, m.match => "([#$id](https://github.com/$url/issues/$id))")
+    end
+    # ## Version `vX.Y.Z` -> ## Version [`vX.Y.Z`](url/releases/tag/vX.Y.Z)
+    while (m = match(r"\#\# Version `v([0-9]+.[0-9]+.[0-9]+)`", line)) !== nothing
+        tag = m.captures[1]
+        line = replace(
+            line,
+            m.match => "## Version [$tag](https://github.com/$url/releases/tag/v$tag)",
+        )
+    end
+    return line
+end
+
+open(joinpath(dirname(@__DIR__), "CHANGELOG.md"), "r") do in_io
+    open(joinpath(@__DIR__, "src", "release_notes.md"), "w") do out_io
+        for line in readlines(in_io; keep = true)
+            write(out_io, fix_release_line(line))
+        end
+    end
+end
```

Comment:
Everything above this is also duplicated in JuMP docs, right? Should we create a `Changelog.jl`?

Reply:
The exact regexes are slightly different. But yeah, we could standardize and then add this as a function to Documenter. I don't think it needs to be a package.

Reply:
You're right, this is quite Documenter-specific right now. We could drop it into Documenter as an undocumented feature right now (we might want to iterate on the exact form of the public API, but let's not do it as part of this PR).

Reply:
I think we can have some standard rewrites:

 * `(abc#XXXX)` -> `([abc#XXXX](github.com/abc/issue/XXXX))`
 * `(#XXXX)` -> `([#XXXX](github.com/default_url/issue/XXXX))`
 * `## vX.Y.Z` -> `## [vX.Y.Z](url/releases/tag/vX.Y.Z)`

JuMP also has

 * `(@abc)` -> `([@abc](github.com/abc))`

The main issue is the formatting, e.g., do we expect `## Version vX.Y.Z`, `## vX.Y.Z` or `## Version vX.Y.Z`?

Reply:
I do kind of like having the issue references working in `CHANGELOG.md` as well though (like we do right now here). Unfortunately, it doesn't look like GitHub has any syntax in Markdown files that makes these links work, so one would need to maintain reference links.

I suppose one approach would be to generate both `CHANGELOG.md` and `release_notes.md` from the same source file (and have CI make sure `CHANGELOG.md` is up to date). The rewriting rules look good to me.

Mainly just thinking about loud here though -- this all adds complexity.

> e.g., do we expect ## Version vX.Y.Z, ## vX.Y.Z or ## Version vX.Y.Z

I think it might be good if the version heading line is a bit longer -- so something like `# Version [v1.1.0](...) - 2019-02-15` seems reasonable to me?

Reply:
Losing the links in CHANGELOG is a downside. But then again... they are in the online documentation which most users would look at.

I considered options like keeping CHANGELOG in the proper format, and having a script that rebuilt it in-place, but that just added more work for the maintainer because you'd have to remember to periodically rebuild it. You might as well just force people writing the changelog to add all the links.

A benefit of this approach is that you can also reference docstrings and stuff in the CHANGELOG because it's now part of Documenter. JuMP does this, so when we add a new function, clicking on it takes the user to the docstring;

![image](https://user-images.githubusercontent.com/8177701/217391039-f8ad9c7a-e2e4-4ce9-97f8-7b6795d405e9.png)

Reply:
Yep, I think that's fair. Let's stick to this approach.